packages/expo/src/hooks/useAppleSignIn.ts

```diff
@@ -0,0 +1,159 @@
+ * between sign-in and sign-up.
+ *
+ * @example
+ * ```tsx
+ * import { useAppleSignIn } from '@clerk/clerk-expo';
+ * import { Button } from 'react-native';
+ *
+ * function AppleSignInButton() {
+ *   const { startAppleSignInFlow } = useAppleSignIn();
+ *
+ *   const onPress = async () => {
+ *     try {
+ *       const { createdSessionId, setActive } = await startAppleSignInFlow();
+ *
+ *       if (createdSessionId && setActive) {
+ *         await setActive({ session: createdSessionId });
+ *       }
+ *     } catch (err) {
+ *       console.error('Apple Sign-In error:', err);
+ *     }
+ *   };
+ *
+ *   return <Button title="Sign in with Apple" onPress={onPress} />;
+ * }
+ * ```
+ *
+ * @requires expo-apple-authentication - Must be installed as a peer dependency
+ * @platform iOS - This hook only works on iOS. On other platforms, it will throw an error.
+ *
+ * @returns An object containing the `startAppleSignInFlow` function
+ */
+export function useAppleSignIn() {
+  const { signIn, setActive, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startAppleSignInFlow(
+    startAppleSignInFlowParams?: StartAppleSignInFlowParams,
+  ): Promise<StartAppleSignInFlowReturnType> {
+    // Check platform compatibility
+    if (Platform.OS !== 'ios') {
+      return errorThrower.throw(
+        'Apple Sign-In is only available on iOS. Please use the web-based OAuth flow (useSSO with strategy: "oauth_apple") on other platforms.',
+      );
+    }
+
+    if (!isSignInLoaded || !isSignUpLoaded) {
+      return {
+        createdSessionId: null,
+        signIn,
+        signUp,
+        setActive,
+      };
+    }
+
+    // Check if Apple Authentication is available on the device
+    const isAvailable = await AppleAuthentication.isAvailableAsync();
+    if (!isAvailable) {
+      return errorThrower.throw('Apple Authentication is not available on this device.');
+    }
+
+    try {
+      // Generate a cryptographically secure nonce for the Apple Sign-In request (required by Clerk)
+      // Lazy load expo-crypto to avoid import issues on web
+      const Crypto = await import('expo-crypto');
+      const nonce = Crypto.randomUUID();
+
+      // Request Apple authentication with requested scopes
+      const credential = await AppleAuthentication.signInAsync({
+        requestedScopes: [
+          AppleAuthentication.AppleAuthenticationScope.FULL_NAME,
+          AppleAuthentication.AppleAuthenticationScope.EMAIL,
+        ],
+        nonce,
+      });
+
+      // Extract the identity token from the credential
+      const { identityToken } = credential;
+
+      if (!identityToken) {
+        return errorThrower.throw('No identity token received from Apple Sign-In.');
+      }
+
+      // Create a SignIn with the Apple ID token strategy
+      // Note: Type assertions needed until @clerk/clerk-react propagates the new oauth_token_apple strategy type
```

Comment:
Since you updated the `types` package we don't need the `as any` cast anymore right?